src/NuGet.Core/NuGet.Configuration/PackageSource/PackageSourceProvider.cs

```diff
@@ -18,17 +18,34 @@ public class PackageSourceProvider : IPackageSourceProvider
 
         public PackageSourceProvider(
           ISettings settings)
-            : this(settings, ConfigurationDefaults.Instance.DefaultPackageSources)
+            : this(settings, ConfigurationDefaults.Instance.DefaultPackageSources, enablePackageSourcesChangedEvent: true)
+        {
+        }
+
+        public PackageSourceProvider(
+          ISettings settings,
+          bool enablePackageSourcesChangedEvent)
+            : this(settings, ConfigurationDefaults.Instance.DefaultPackageSources, enablePackageSourcesChangedEvent)
         {
         }
 
         public PackageSourceProvider(
             ISettings settings,
-            IEnumerable<PackageSource> configurationDefaultSources
-            )
+            IEnumerable<PackageSource> configurationDefaultSources)
+            : this (settings, configurationDefaultSources, enablePackageSourcesChangedEvent: true)
+        {
+        }
+
+        public PackageSourceProvider(
+            ISettings settings,
+            IEnumerable<PackageSource> configurationDefaultSources,
+            bool enablePackageSourcesChangedEvent)
         {
             Settings = settings ?? throw new ArgumentNullException(nameof(settings));
-            Settings.SettingsChanged += (_, __) => { OnPackageSourcesChanged(); };
+            if (enablePackageSourcesChangedEvent)
+            {
+                Settings.SettingsChanged += (_, __) => { OnPackageSourcesChanged(); };
```

Comment:
When is this disposed of?

Reply:
it's not. NuGet/Home#8479 needs to fix it. We can't fix it without breaking backwards compat for customers using our package.